### PR TITLE
doc: build files: move the list to new page

### PR DESCRIPTION
**Pull request by greg-fer**
Moved the reference list of output build files to a new page in the Configuration and building section. Edited other pages for links.

Removed Building FOTA images section and edited info in tables and cross-links.

Added link to MCUboot build output files in Adding MCUboot as an upgradable bootloader.

NCSDK-22324 and NCSIDB-964.

### CODE REVIEW

File: doc/nrf/config_and_build/output_build_files.rst [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+Output build files (image files)
+################################
+
+The building process produces each time an :term:`image file`.
+
+The image file can refer to an *executable*, a *program*, or an *ELF file*.
+As one of the last build steps, the linker processes all object files by locating code, data, and symbols in sections in the final ELF file.
+The linker replaces all symbol references to code and data with addresses.
+A symbol table is created which maps addresses to symbol names, which is used by debuggers.
```
Comment (peknis) [inline doc/nrf/config_and_build/output_build_files.rst:11]:
```suggestion
A symbol table that maps addresses to symbol names is created.
This table is used by debuggers.
```

File: doc/nrf/config_and_build/output_build_files.rst [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+When an ELF file is converted into another format, such as HEX or binary, the symbol table is lost.
```
Comment (peknis) [inline doc/nrf/config_and_build/output_build_files.rst:12]:
```suggestion
When an ELF file is converted to another format, such as HEX or binary, the symbol table is lost.
```

File: doc/nrf/config_and_build/output_build_files.rst [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+The |NCS| build system places output images in the :file:`<build folder>/zephyr` folder.
+
+.. _app_build_output_files_common:
+
+Common output build files
+*************************
+
+The following table lists build files that can be generated as output when building firmware for supported :ref:`build targets <app_boards>`.
+The table includes files for single-core and multi-core programming scenarios for both |VSC| and command line building methods.
```
Comment (peknis) [inline doc/nrf/config_and_build/output_build_files.rst:24]:
```suggestion
The table includes files for single-core and multi-core programming scenarios for both |VSC| and command-line building methods.
```

File: doc/nrf/config_and_build/output_build_files.rst [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+Which files you are going to use depends on the application configuration and not directly on the type of SoC you are using.
+The following scenarios are possible:
+
+* Single-image - Only one firmware image file is generated for a single core.
+* Multi-image - Two or more firmware image files are generated for a single core.
+  You can read more about this scenario in :ref:`ug_multi_image`.
```
Comment (peknis) [inline doc/nrf/config_and_build/output_build_files.rst:30]:
```suggestion
  For more details about this scenario, see :ref:`ug_multi_image`.
```

File: doc/nrf/config_and_build/output_build_files.rst [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+|                                          | For example: :file:`127F-0141-01020003-light_switch.zigbee`.                                          |                                                                                                           |
++------------------------------------------+-------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------+
+
+.. _app_build_mcuboot_output:
+
+MCUboot output build files
+**************************
+
+.. note::
+    MCUboot's :doc:`direct-xip mode <mcuboot:design>` and the related :kconfig:option:`CONFIG_BOOT_BUILD_DIRECT_XIP_VARIANT` Kconfig option are currently supported only on the single-core devices such as the nRF52 Series (:ref:`more information <ug_nrf52_developing_ble_fota_mcuboot_direct_xip_mode>`).
```
Comment (peknis) [inline doc/nrf/config_and_build/output_build_files.rst:93]:
```suggestion
.. note::
   MCUboot's :doc:`direct-xip mode <mcuboot:design>` and the related :kconfig:option:`CONFIG_BOOT_BUILD_DIRECT_XIP_VARIANT` Kconfig option are currently supported only on single-core devices, such as nRF52 Series devices.
   For more details, see the :ref:`ug_nrf52_developing_ble_fota_mcuboot_direct_xip_mode` section of the :ref:`ug_nrf52_developing` page.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by hellesvik-nordic — APPROVED** [review]

**Review by peknis — APPROVED** [review]

**Review by MarekPieta — COMMENTED** [review]
Post merge review. Looks good :)

**Pull request merged**